source/constructs/api/data_source/service.py

```diff
@@ -2581,29 +2673,356 @@ def query_connection_detail(account: JDBCInstanceSourceBase):
+    __s3_client.upload_fileobj(excel_bytes, admin_bucket_name, batch_create_ds)
+    return f'report_{time_str}'
+
+def __check_empty_for_field(row, header):
+    if row[0].value is None or str(row[0].value).strip() == const.EMPTY_STR:
+        return True, f"{header[0]} should not be empty"
+    if row[1].value is None or str(row[1].value).strip() == const.EMPTY_STR:
+        return True, f"{header[1]} should not be empty"
+    if row[3].value is None or str(row[3].value).strip() == const.EMPTY_STR:
+        return True, f"{header[3]} should not be empty"
+    if row[5].value is None or str(row[5].value).strip() == const.EMPTY_STR:
+        if row[6].value is None or str(row[6].value).strip() == const.EMPTY_STR:
+            return True, f"{header[6]} should not be empty when {header[5]} is empty"
+        if row[7].value is None or str(row[7].value).strip() == const.EMPTY_STR:
+            return True, f"{header[7]} should not be empty when {header[5]} is empty"
+    if row[8].value is None or str(row[8].value).strip() == const.EMPTY_STR:
+        return True, f"{header[8]} should not be empty"
+    if row[9].value is None or str(row[9].value).strip() == const.EMPTY_STR:
+        return True, f"{header[9]} should not be empty"
+    if row[10].value is None or str(row[10].value).strip() == const.EMPTY_STR:
+        return True, f"{header[10]} should not be empty"
+    return False, None
+
+def __map_network_jdbc(created_jdbc_list, subnets, security_group_id):
+    res = []
+    for index, item in enumerate(created_jdbc_list):
+        item.network_sg_id = security_group_id
+        item.network_subnet_id = subnets[0] if index % 2 == 0 else subnets[1]
+        res.append(item)
+    return res
 
-def list_jdbc_databases(source: JdbcSource) -> list[str]:
-    url_arr = source.connection_url.split(":")
-    if len(url_arr) != 4:
-        raise BizException(MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_code(), MessageEnum.SOURCE_JDBC_URL_FORMAT_ERROR.get_msg())
-    if url_arr[1] != "mysql":
-        raise BizException(MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_code(), MessageEnum.SOURCE_JDBC_LIST_DATABASES_NOT_SUPPORTED.get_msg())
-    host = url_arr[2][2:]
-    port = int(url_arr[3].split("/")[0])
-    user = source.username
-    password = source.password
-    if source.secret_id:
-        secrets_client = boto3.client('secretsmanager')
-        secret_response = secrets_client.get_secret_value(SecretId=source.secret_id)
-        secrets = json.loads(secret_response['SecretString'])
-        user = secrets['username']
-        password = secrets['password']
-    mysql_database = jdbc_database.MySQLDatabase(host, port, user, password)
-    databases = mysql_database.list_databases()
-    logger.info(databases)
-    return databases
+def query_batch_status(filename: str):
+    success, warning, failed = 0, 0, 0
+    file_key = f"{const.BATCH_CREATE_REPORT_PATH}/{filename}.xlsx"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;filename&#39; (index: 0 | type: str). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other methods validating the parameter: &#39;download_batch_file&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).